### PR TITLE
test(typer/swift): add track edge-case validator coverage

### PR DESCRIPTION
**Pull request by dhawal1248**
## 🔗 Ticket

Resolves [DEX-322](https://linear.app/rudderstack/issue/DEX-322)

Parent: [DEX-311](https://linear.app/rudderstack/issue/DEX-311) — Add e2e tests for RudderTyper Swift.
Stacks on: [DEX-321 / PR #518](https://github.com/rudderlabs/rudder-iac/pull/518) — core track coverage. Base branch is set to that PR so GitHub shows only the DEX-322 delta; once the parents merge in order this retargets to `main` automatically.

Completes the `testIdentify` → `testGroup/Screen` → core-track → edge-track stack and closes out the DEX-311 initiative on the tests side.

---

## Summary

Adds six XCTest cases covering the generator's less-obvious track shapes — variant discriminators, primitive enum edges, and empty-struct serialization. Each test's header comment explicitly calls out the Swift-specific divergence from the Kotlin suite so a reader doesn't mistake the difference for a bug.

---

## Changes

- **`testTrackEventWithVariants`** — asserts the top-level variants discriminator routes the right payload case through the generated enum (the `Event With Variants` event's `.caseMobile` shape).
- **`testTrackUserSignedUpFeatureConfigs`** — exercises all four `feature_config` cases (`true` / `false` / `beta` / default). The Swift generator emits an empty `Default` struct, so the default case serializes to `{}`. The Kotlin suite's `Default` carries `featureFlag` — the divergence is called out inline.
- **`testTrackUserSignedUpEnums`** — asserts primitive enum serialization. `PropertyEnabled` and `PropertyMixedValue` are String-backed in Swift (rawValue `"true"` / `"2.5"`), so the serialized JSON carries strings rather than the booleans / numbers the Kotlin test asserts. Called out in the header comment.
- **`testTrackUserSignedUpUserAccess`** — `CustomTypeUserAccess.Default` has no fields in Swift, so `user_access` serializes to `{}` rather than Kotlin's `{"active": …}`.
- **`testTrackUserSignedUpAllEnums`** — combined-enums sanity check with every enum variant populated at once, guarding against cross-talk between the generated raw-value mappings.
- **`testTrackUserSignedUpUnitSerialization`** — empty-object property. Swift models `nestedEmptyObjectNoAdditionalProps` as `[String: Any]?` (Kotlin models it as `Unit`), so the test passes `[:]` and asserts the key is serialized as `{}`.

---

## Testing

- `make typer-swift-validate` runs all 12 tests (this set + identify + group + screen + the three core track cases) stacked on DEX-319 / DEX-320 / DEX-321 + DEX-317's testdata fix: all pass.
- No Go code or generator behaviour changes.

---

## Risk / Impact

Low.

Test-only addition gated behind the opt-in `make typer-swift-validate` target. No production code paths touched; the Swift generator itself is unchanged.

---

## Checklist

- [x] Ticket linked
- [x] Tests added/updated
- [x] No breaking changes (or documented)

### CODE REVIEW

File: cli/internal/typer/generator/platforms/swift/testdata/validator/Sources/RudderTyperTests/RudderTyperTests.swift [hunk 1]
```diff
@@ -255,4 +255,209 @@ final class RudderTyperTests: XCTestCase {
...
+            "active": true,
+            "feature_config": [:] as [String: Any],
+            "profile": ["email": "feature.alpha@example.com", "first_name": "Alpha", "last_name": "User"],
+        ]))
+    }
+
+    // MARK: - track (primitive enums)
+    //
+    // Swift emits `PropertyEnabled` and `PropertyMixedValue` as String-backed
+    // enums (rawValue is a String like "true" / "2.5"), so the serialized JSON
```
Comment (akashrpo) [inline cli/internal/typer/generator/platforms/swift/testdata/validator/Sources/RudderTyperTests/RudderTyperTests.swift:345]: Should enabled and mixed_value be serialized as strings ("true", "2.5") instead of real booleans/numbers? Is this intended?

**Review by akashrpo — APPROVED** [review]
